easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -331,9 +332,61 @@ def configure_step(self, srcdir=None, builddir=None):
                 self.cfg['configopts']])
 
         res = run_shell_cmd(command)
-
+        self.check_python_paths()
         return res.output
 
+    def check_python_paths(self):
+        """Check that there are no detected Python paths outside the Python dependency provided by EasyBuild"""
+        if not os.path.exists('CMakeCache.txt'):
+            self.log.warning("CMakeCache.txt not found. Python paths checks skipped.")
+            return
+        cmake_cache = read_file('CMakeCache.txt')
+        if not cmake_cache:
+            self.log.warning("CMake Cache could not be read. Python paths checks skipped.")
+            return
+
+        self.log.info("Checking Python paths")
+
+        python_paths = {
+            "executable": [],
+            "include_dir": [],
+            "library": [],
+        }
+        python_regex = re.compile(r"_?(?:Python|PYTHON)\d?_(EXECUTABLE|INCLUDE_DIR|LIBRARY)[^=]*=(.*)")
```

Comment:
The [^=]* part matches to much. From my test builds, I have things like
```
PYTHON_EXECUTABLE-ADVANCED:INTERNAL=1
```

Reply:
maybe sufficient to just add a : ?

```suggetion
        python_regex = re.compile(r"_?(?:Python|PYTHON)\d?_(EXECUTABLE|INCLUDE_DIR|LIBRARY):[^=]*=(.*)")
```

Reply:
That is not enough to match variables with a potential suffixes like "_DIR" or "_DEBUG", like `PYTHON_LIBRARY_DEBUG`. However limiting to word characters should work. See [the](https://github.com/easybuilders/easybuild-easyblocks/pull/3399/commits/23451116f6395cc34268a594f4ab56ad69cd001a) where I also added some examples in the commit message